tests/test_components.py

```diff
@@ -21,3 +32,45 @@ def test_metadata_default_values():
     metadata = Metadata()
     assert metadata.dateDownloaded is None
     assert metadata.probability is None
+
+
+def test_anchor_optional_fields():
+    Link(text="foo")
+    Link(url="https://example.com")
+
+
+def test_business_place_metadata_default_values():
+    metadata = BusinessPlaceMetadata()
+    assert metadata.dateDownloaded is None
+    assert metadata.probability is None
+    assert metadata.searchText is None
+
+
+def test_address():
+    Address(addressRaw="2440 Hoonani Rd, Koloa, HI 96756, US")
+    Address(streetAddress="2440 Hoonani Rd, Koloa, HI 96756")
+    Address(addressCity="Koloa")
+    Address(addressLocality="Poipu")
+    Address(addressRegion="Kauaʻi County")
+    Address(addressCountry="US")
+    Address(postalCode="HI 96756")
+    Address(latitude=21.8768183)
+    Address(longitude=-159.3620652)
+
+
+def test_amenity():
+    Amenity(name="WiFi", value=True)
+    Amenity(name="WiFi", value=False)
+
+
+def test_star_rating():
+    StarRating(raw="4-star hotel")
+    StarRating(ratingValue=4)
+
+
+def test_parent_place():
+    ParentPlace(name="Gateway Plaza", placeId="ChIJCVDfcEKuEmsR7_zYc0_GBtQ")
+
+
+def test_opening_hours():
+    OpeningHours(dayOfWeek="Sunday", opens="9:00", closes="12:00")
```

Comment:
I wonder if we should find a name that does not match the array property where this would be but 1 item. `OpeningHoursItem`? `OpeningHoursPeriod`? `OpeningPeriod`?